src/components/common/FilterListbox.tsx

```diff
@@ -0,0 +1,77 @@
+import { Listbox } from "@headlessui/react";
+import { ChevronDown } from "lucide-react";
+import { Fragment } from "react";
+import { cn } from "@/lib/utils";
+export interface FilterOption {
+  value: string;
+  sort: string;
+}
+
+interface FilterListboxProps {
+  value: string;
+  onChange: (value: string) => void;
+  options: Record<string, FilterOption> | FilterOption[];
+  buttonClassName?: string;
+  width?: string;
+}
+
+export default function FilterListbox({
+  value,
+  onChange,
+  options,
+  buttonClassName = "bg-wash",
+  width = "w-[176px]",
+}: FilterListboxProps) {
+  const optionsArray = Array.isArray(options)
+    ? options
+    : Object.values(options);
+
+  const getCurrentLabel = () => {
+    if (Array.isArray(options)) {
+      return options.find((opt) => opt.sort === value)?.value;
+    }
+    return (
+      options[value]?.value ||
+      Object.values(options).find((opt) => opt.sort === value)?.value
+    );
+  };
+
+  return (
+    <Listbox as="div" value={value} onChange={onChange}>
+      {() => (
```

Comment:
Nit: Don't think this is needed here, if you update the PR for any other changes, consider removing this. Though its fine for now, it can break memoisation if we pass any arrow functions later into these.